hr_announcement/models/announcement.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class Announcement(models.Model):
+    _inherit = "announcement"
+
+    announcement_type = fields.Selection(
+        selection_add=[
+            ("employee", "Employee"),
+            ("department", "Department"),
+            ("job_position", "Job Position"),
+        ],
+        ondelete={
+            "employee": "set default",
+            "department": "set default",
+            "job_position": "set default",
+        },
+    )
+    employee_ids = fields.Many2many(
+        comodel_name="hr.employee",
+        string="Employees",
+    )
+    department_ids = fields.Many2many(
+        comodel_name="hr.department",
+        string="Departments",
+    )
+    position_ids = fields.Many2many(
+        comodel_name="hr.job",
+        string="Job Positions",
+    )
+
+    def _update_read_unread_announcements(self):
+        """Used to set users unread announcements when they're set in the announcement
+        itself"""
+        for announcement in self:
+            for employee in announcement.employee_ids.filtered(
+                lambda x: x.user_id
+                and announcement
+                not in (
+                    x.user_id.read_announcement_ids + x.user_id.unread_announcement_ids
+                )
+            ):
+                employee.user_id.unread_announcement_ids |= announcement
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        announcements = super().create(vals_list)
+        for vals in vals_list:
+            if vals.get("employee_ids"):
+                announcements._update_read_unread_announcements()
```

Comment:
By using `model_create_multi` you would be running it for all announcements with only 1 value having `employee_ids`.

```suggestion
        announcements = super().create(vals_list)
        for index, vals in enumerate(vals_list):
            if vals.get("employee_ids"):
                announcements[index]._update_read_unread_announcements()
```

Or you can directly use `@api.model` in a unitary way.

Reply:
thanksI still have a lot to learn but looking at the documentation I have learned the difference between both decorators, thanks.

Reply:
Tranquila, si en este mundo estamos todos aprendiendo continuamente.